despatch_advice_import/wizard/despatch_advice_import.py

```diff
@@ -0,0 +1,297 @@
+    # 'despatch_advice_type_code': ' scheduled | delivered'
+    # 'supplier': {'vat': 'FR25499247138'},
+    # 'company': {'vat': 'FR12123456789'}, # Only used to check we are not
+    #                                      # importing the quote in the
+    #                                      # wrong company by mistake
+    # 'estimated_delivery_date': '2020-11-20'
+    # 'lines': [{
+    #           'id': 123456,
+    #           'qty': 2.5,
+    #           'uom': {'unece_code': 'C62'},
+    #           'backorder_qty: None  # if provided and qty != expected
+    #                                 # the backorder qty will be delivered
+    #                                 # in a next shipping
+    #    }]
+
+    @api.model
+    def parse_despatch_advice(self, document, filename):
+        if not document:
+            raise UserError(_("Missing document file"))
+        if not filename:
+            raise UserError(_("Missing document filename"))
+        filetype = mimetypes.guess_type(filename)[0]
+        logger.debug("DespatchAdvice file mimetype: %s", filetype)
+        if filetype in ["application/xml", "text/xml"]:
+            try:
+                xml_root = etree.fromstring(document)
+            except Exception as err:
+                raise UserError(_("This XML file is not XML-compliant")) from err
+            if logger.isEnabledFor(logging.DEBUG):
+                pretty_xml_string = etree.tostring(
+                    xml_root, pretty_print=True, encoding="UTF-8", xml_declaration=True
+                )
+                logger.debug("Starting to import the following XML file:")
+                logger.debug(pretty_xml_string)
+            parsed_despatch_advice = self.parse_xml_despatch_advice(xml_root)
+        elif filetype == "application/pdf":
+            parsed_despatch_advice = self.parse_pdf_despatch_advice(document)
+        else:
+            raise UserError(
+                _(
+                    "This file '%s' is not recognised as XML nor PDF file. "
+                    "Please check the file and it's extension."
+                )
+                % filename
+            )
+        logger.debug("Result of Despatch Advice parsing: ", parsed_despatch_advice)
+        if "attachments" not in parsed_despatch_advice:
+            parsed_despatch_advice["attachments"] = {}
+        parsed_despatch_advice["attachments"][filename] = b64encode(document)
+        if "chatter_msg" not in parsed_despatch_advice:
+            parsed_despatch_advice["chatter_msg"] = []
+        if parsed_despatch_advice.get("company") and not self.env.context.get(
+            "edi_skip_company_check"
+        ):
+            self.env["business.document.import"]._check_company(
+                parsed_despatch_advice["company"], parsed_despatch_advice["chatter_msg"]
+            )
+        defaults = self.env.context.get("despatch_advice_import__default_vals", {}).get(
```

Comment:
hmm this part seems not correct. We should have a lookup for the picking defaults and in the line handling a lookup for lines' defaults

Reply:
https://github.com/OCA/edi/blob/14.0/sale_order_import/wizard/sale_order_import.py#L271

https://github.com/OCA/edi/blob/14.0/sale_order_import/wizard/sale_order_import.py#L486

Reply:
My lines part is here: https://github.com/OCA/edi/blob/b72af6a075cde1156d907e37436bde8531cc47f0/despatch_advice_import_ubl/wizard/despatch_advice_import.py#L128-L129
I think you don't see it, right?